packages/next/src/server/web/spec-extension/adapters/request-cookies.ts

```diff
@@ -61,6 +62,14 @@ export class MutableRequestCookiesAdapter {
     let modifiedValues: ResponseCookie[] = []
     const modifiedCookies = new Set<string>()
     const updateResponseCookies = () => {
+      // TODO-APP: change method of getting staticGenerationAsyncStore
+      const staticGenerationAsyncStore = (fetch as any)
+        .__nextGetStaticStore?.()
+        ?.getStore() as undefined | StaticGenerationStore
+      if (staticGenerationAsyncStore) {
+        staticGenerationAsyncStore.pathWasRevalidated = true
+      }
+
```

Comment:
I think we still need to make sure `actionAsyncStorage.getStore().isAction === true` here as `cookies().set()` can happen inside Route Handlers too.

Reply:
this property doesn't do anything outside of server actions